File: src/main/java/mediathek/gui/actions/ManageBookmarkAction.java
package mediathek.gui.actions;

import mediathek.mainwindow.MediathekGui;
import mediathek.swing.IconUtils;
import org.kordamp.ikonli.materialdesign2.MaterialDesignF;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class ManageBookmarkAction extends AbstractAction {
    private final MediathekGui mediathekGui;

    public ManageBookmarkAction(MediathekGui mediathekGui) {
        this.mediathekGui = mediathekGui;
        putValue(Action.NAME, "Merkliste verwalten...");
        putValue(Action.SMALL_ICON, IconUtils.toolbarIcon(MaterialDesignF.FILE_DOCUMENT));
        putValue(Action.SHORT_DESCRIPTION, "Merkliste verwalten");
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        mediathekGui.tabFilme.showManageBookmarkWindow();
    }
}

File: src/main/kotlin/mediathek/gui/actions/ManageBookmarkAction.kt
package mediathek.gui.actions

import mediathek.mainwindow.FilmBookmarkHost
import mediathek.swing.IconUtils
import org.kordamp.ikonli.materialdesign2.MaterialDesignF
import java.awt.event.ActionEvent
import javax.swing.AbstractAction

class ManageBookmarkAction(
    private val host: FilmBookmarkHost,
) : AbstractAction() {
    init {
        putValue(NAME, "Merkliste verwalten...")
        putValue(SMALL_ICON, IconUtils.toolbarIcon(MaterialDesignF.FILE_DOCUMENT))
        putValue(SHORT_DESCRIPTION, "Merkliste verwalten")
    }

    override fun actionPerformed(event: ActionEvent?) {
        host.showManageBookmarkWindow()
    }
}
